pyro/optim/optim.py

```diff
@@ -51,12 +51,12 @@ def __call__(self, params,  *args, **kwargs):
                     state = self._state_waiting_to_be_consumed.pop(param_name)
                     self.optim_objs[p].load_state_dict(state)
 
-            # actually perform the step for the optim object
-            self.optim_objs[p].step(*args, **kwargs)
-
-            # if optim object was a scheduler, perform an actual optim step
-            if isinstance(self.optim_objs[p], torch.optim.lr_scheduler._LRScheduler):
+            if isinstance(self.optim_objs[p], torch.optim.lr_scheduler._LRScheduler) or \
```

Comment:
smells bad to me.

Reply:
i'm not sure how to make it cleaner given the design choice. we just want to step the optimizers here.  `optim_objs` is a dict of optimizers _or_ a dict of schedulers each of which is wrapping an optimizer.  in the `lr_scheduler` initialization, we could store the schedulers and optimizers separately, which would simplify this logic but essentially have two references to the optimizer.  WDYT?

Reply:
ok, I guess we've inherited this bad smell from PyTorch 😞